video/out/present_sync.c

```diff
@@ -37,48 +38,104 @@
 
 void present_sync_get_info(struct mp_present *present, struct vo_vsync_info *info)
 {
-    info->vsync_duration = present->vsync_duration;
-    info->skipped_vsyncs = present->last_skipped_vsyncs;
-    info->last_queue_display_time = present->last_queue_display_time;
+    struct mp_present_entry *cur = present->head;
+    while (cur) {
+        if (cur->queue_display_time)
+            break;
+        cur = cur->list_node.next;
+    }
+    if (!cur)
+        return;
+
+    info->vsync_duration = cur->vsync_duration;
+    info->skipped_vsyncs = cur->skipped_vsyncs;
+    info->last_queue_display_time = cur->queue_display_time;
+
+    // Remove from the list, zero out everything, and append at the end
+    LL_REMOVE(list_node, present, cur);
+    cur->ust = 0;
+    cur->msc = 0;
+    cur->vsync_duration = 0;
+    cur->skipped_vsyncs = 0;
+    cur->queue_display_time = 0;
```

Comment:
nitpick, but this would be more concise

```suggestion
    *cur = (mp_present_entry){0};
```